### ISSUE TITLE
[Enhancement] Simplified storage layer Timer

### ISSUE THREAD

**Issue by wangbo** [apache/doris#8837]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description
In Doris, we usually use Timer to observe performance.
But Timer itself also brings cost.
For exmaple,
```
sql:

SELECT sum(LO_EXTENDEDPRICE * LO_DISCOUNT) FROM test.lineorder_flat WHERE C_REGION = 'AMERICA'

SegmentIterator:
           - BitmapIndexFilterTimer: 287.451us
           - BlockLoadTime: 10s933ms
           - BlockSeekCount: 11.99M
           - BlockSeekTime: 0.000ns
           - BlocksLoad: 73.50K
```
We can see that this sql may bring too many seekCount, this because the predicate ```C_REGION = 'AMERICA'``` has poor selectivity.
This means BlockSeekTimer may cost much.
The origin perf as below:
<img width="520" alt="1" src="https://user-images.githubusercontent.com/12951030/161416949-65f3abde-5962-476d-af73-a44f0c2f9259.png">

Test 1,remove some Timer in next batch
<img width="520" alt="2" src="https://user-images.githubusercontent.com/12951030/161416970-e5de2aa0-fa23-4395-96bf-148e89a24891.png">
We can see cost dropped a little.
This may because ```next batch``` is called on the order of 10,000.

Test 2, remove Timer in seek
<img width="520" alt="3" src="https://user-images.githubusercontent.com/12951030/161417036-0dfff9e9-8012-4877-8c07-56be41022ce1.png">
We can see that about 7% cost is eliminated.

I also test other sql like SSB, Q2,3,4, the timer cost not so much, about 1.59% .
Because that SQL cost more time in other places.
For a fast SQL, Timer cost will account for a larger proportion.

In conclusion:
1 Timer also bring cost. Especially in case of mixed load of Prod Env.
2 We should not make timer in hot code path.
3 Storage Layer Timer can be simplified

### Solution

I think Timer in Storage Layer can be divided to three parts.
1 Read Timer.
2 Evaluate Timer.
3 OutputTimer.

I will make a PR after #8794 is merged.
### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)
